



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.1.0">
    
    
      
        <title>Roary - Sepsis Training 2016</title>
      
    
    
      <link rel="stylesheet" href="../../../../assets/stylesheets/application.982221ab.css">
      
      
    
    
      <script src="../../../../assets/javascripts/modernizr.01ccdecf.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../../../css/sepsis.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#roary" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../../.." title="Sepsis Training 2016" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Sepsis Training 2016
            </span>
            <span class="md-header-nav__topic">
              Roary
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../../.." title="Sepsis Training 2016" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Sepsis Training 2016
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Microbial Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Microbial Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/galaxy/" title="Starting with Galaxy" class="md-nav__link">
      Starting with Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/data-dna/" title="Training dataset" class="md-nav__link">
      Training dataset
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/velvet/" title="Genome assembly with Velvet" class="md-nav__link">
      Genome assembly with Velvet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/spades/" title="Genome assembly with Spades" class="md-nav__link">
      Genome assembly with Spades
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/variants/" title="Basic Variant finding" class="md-nav__link">
      Basic Variant finding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/snippy/" title="Variant finding" class="md-nav__link">
      Variant finding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Advanced Microbial Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Advanced Microbial Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/pacbio/" title="Assembly with SMRT portal" class="md-nav__link">
      Assembly with SMRT portal
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Microbial Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Microbial Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../../modules/dge/" title="Differential gene expression" class="md-nav__link">
      Differential gene expression
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../about/" title="About" class="md-nav__link">
      About
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#pre-requisites" title="Pre-requisites" class="md-nav__link">
    Pre-requisites
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#start" title="Start" class="md-nav__link">
    Start
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input-data" title="Input data" class="md-nav__link">
    Input data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#run" title="Run" class="md-nav__link">
    Run
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-it-works" title="How it works" class="md-nav__link">
    How it works
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output" title="Output" class="md-nav__link">
    Output
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#summary-statistics" title="summary statistics:" class="md-nav__link">
    summary statistics:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-presenceabsence-graphically" title="gene presence/absence graphically:" class="md-nav__link">
    gene presence/absence graphically:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#list-of-genes-that-are-presentabsent" title="list of genes that are present/absent:" class="md-nav__link">
    list of genes that are present/absent:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-the-pan-genome" title="query the pan genome:" class="md-nav__link">
    query the pan genome:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#advanced-options" title="Advanced options" class="md-nav__link">
    Advanced options
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-roary-and-create-an-alignment-of-core-genes" title="Run roary and create an alignment of core genes:" class="md-nav__link">
    Run roary and create an alignment of core genes:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#generate-a-tree-based-on-the-presenceabsence-of-core-genes" title="Generate a tree based on the presence/absence of core genes:" class="md-nav__link">
    Generate a tree based on the presence/absence of core genes:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#use-roary_plotspy-to-generate-plots" title="Use roary_plots.py to generate plots:" class="md-nav__link">
    Use roary_plots.py to generate plots:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-next" title="What next?" class="md-nav__link">
    What next?
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#more-information" title="More information" class="md-nav__link">
    More information
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="roary">Roary</h1>
<p>cmdline</p>
<p>This tutorial demonstrates how to calculate the pan and core genomes of a set of input bacterial species, using Roary.</p>
<p><a href="https://sanger-pathogens.github.io/Roary/">Roary code and manual on github</a></p>
<p><a href="http://bioinformatics.oxfordjournals.org/content/31/22/3691">Roary paper</a></p>
<h2 id="pre-requisites">Pre-requisites</h2>
<ul>
<li>background: pan genomes</li>
<li>a mGVL instance</li>
</ul>
<h2 id="start">Start</h2>
<ul>
<li><strong>via local Terminal</strong>: in your terminal, ssh into your mGVL, but make sure you put in -X -Y after ssh so that xquartz can view files later. (FIXME: word better)</li>
<li><strong>or via virtual desktop</strong>: Go to your mGVL dashboard. Click on the link to the Lubuntu desktop. A virtual desktop will open in a new browser window. Enter username: ubunutu; and your GVL password. Click on terminal in the top left corner.</li>
</ul>
<h2 id="input-data">Input data</h2>
<ul>
<li>Roary takes .gff files produced by Prokka. A gff file has sequences and annotations.</li>
<li><a href="../dna/anno/prokka">file formats</a> FIXME: link to proper page</li>
<li>Get files into mGVL. (FIXME: explain how, wget etc or from GenomeSpace)</li>
<li>put all gff files into a folder</li>
<li>FIXME: choose a good sample set - ideally something that usefully shows how AMR genes can be present/absence in a group? for a draft can use the Listeria tutorial at https://github.com/microgenomics/tutorials/blob/master/pangenome.md</li>
</ul>
<h2 id="run">Run</h2>
<ul>
<li>navigate into the place where the gff folder is.</li>
</ul>
<pre><code class="bash">roary -f ./results ./gff_files/*.gff
</code></pre>

<ul>
<li>&ldquo;-f ./results&rdquo; puts the output into a directory called results</li>
</ul>
<h2 id="how-it-works">How it works</h2>
<ul>
<li>Based on the input genomes, Roary works out which genes are shared between all (core) and which are not (accessory).</li>
<li>It uses the protein-coding genes from each of the input genomes.</li>
<li>converts to protein seqs</li>
<li>similar protein seqs are clustered progressively.</li>
<li>each sample: will be labelled with presence/absence of orthologous genes.</li>
</ul>
<h2 id="output">Output</h2>
<h3 id="summary-statistics">summary statistics:</h3>
<pre><code>more summary_statistics.txt
</code></pre>

<ul>
<li>you will see the number of core genes, shell genes, etc.</li>
<li><code>q</code> to exit viewing</li>
</ul>
<h3 id="gene-presenceabsence-graphically">gene presence/absence graphically:</h3>
<pre><code>roary2svg.pl gene_presence_absence.csv &gt; pan_genome.svg
</code></pre>

<ul>
<li>(if you have logged in with -X -Y)</li>
</ul>
<pre><code>firefox pan_genome.svg &amp;
</code></pre>

<ul>
<li>then <code>enter</code></li>
<li>the &amp; makes it run in the background</li>
<li>a firefox window should open with the svg image</li>
<li>(later: close the firefox window to stop this job running in the background)</li>
</ul>
<h3 id="list-of-genes-that-are-presentabsent">list of genes that are present/absent:</h3>
<ul>
<li>view the gene_presence_absence.csv by (FIXME)</li>
<li>lots of information about this file in the roary website (FIXME summarize?)</li>
</ul>
<h3 id="query-the-pan-genome">query the pan genome:</h3>
<ul>
<li>copy the input .gff files into the results folder (FIXME: do this earlier)</li>
<li>cd into this folder</li>
</ul>
<pre><code>query_pan_genome -a intersection *.gff
</code></pre>

<ul>
<li>this finds the core genes</li>
</ul>
<pre><code>more pan_genome_results
</code></pre>

<ul>
<li>shows the list of genes found in the core genome.</li>
<li><code>q</code> to exit viewing</li>
</ul>
<h2 id="advanced-options">Advanced options</h2>
<p>FIXME: update firefox on mGVL so can run phandango</p>
<h3 id="run-roary-and-create-an-alignment-of-core-genes">Run roary and create an alignment of core genes:</h3>
<pre><code class="bash">roary -f ./results -e -n -p 8 ./gff_files/*.gff
</code></pre>

<ul>
<li>&ldquo;-f ./results&rdquo; puts the output into a directory called results</li>
<li>&ldquo;-e -n&rdquo; creates an alignment of core genes using mafft</li>
<li>&ldquo;-p 8&rdquo; gives 8 threads - optional, if you know how many you have</li>
</ul>
<h3 id="generate-a-tree-based-on-the-presenceabsence-of-core-genes">Generate a tree based on the presence/absence of core genes:</h3>
<ul>
<li>navigate into the results folder that you want to use.</li>
</ul>
<pre><code class="bash">FastTree -nt -gtr core_gene_alignment.aln &gt; my_tree.newick
</code></pre>

<ul>
<li>(By default, roary will also have created a (very quick) tree from the accessory genes.)</li>
<li><a href="http://meta.microbesonline.org/fasttree/">FastTree information and options</a>.</li>
</ul>
<h3 id="use-roary_plotspy-to-generate-plots">Use roary_plots.py to generate plots:</h3>
<ul>
<li>navigate into the results folder that you want to use.</li>
</ul>
<pre><code>python roary_plots.py core_gene_alignment.nwk gene_presence_absence.csv
</code></pre>

<ul>
<li>output: pangenome matrix, frequency plot, pie chart.</li>
<li>view these by typing <code>firefox [filename]</code> and a firefox window will open to show the image. You need to close the window before you open the next image.</li>
</ul>
<h2 id="what-next">What next?</h2>
<p>View using Phandango; <a href="../../../viz/phandango/">tutorial here.</a></p>
<h2 id="more-information">More information</h2>
<ul>
<li><a href="https://github.com/microgenomics/tutorials/blob/master/pangenome.md">another Roary tutorial</a></li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../../assets/javascripts/application.2ba8dec4.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../../../.."}})</script>
      
    
  </body>
</html>