



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.1.0">
    
    
      
        <title>Genome assembly with Velvet - Sepsis Training 2016</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.982221ab.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.01ccdecf.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#assembly-using-velvet" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../.." title="Sepsis Training 2016" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Sepsis Training 2016
            </span>
            <span class="md-header-nav__topic">
              Genome assembly with Velvet
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../.." title="Sepsis Training 2016" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Sepsis Training 2016
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Microbial Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Microbial Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../galaxy/" title="Starting with Galaxy" class="md-nav__link">
      Starting with Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../data-dna/" title="Training dataset" class="md-nav__link">
      Training dataset
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Genome assembly with Velvet
      </label>
    
    <a href="./" title="Genome assembly with Velvet" class="md-nav__link md-nav__link--active">
      Genome assembly with Velvet
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#galaxy-background" title="Galaxy Background" class="md-nav__link">
    Galaxy Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#register-in-galaxy" title="Register in Galaxy" class="md-nav__link">
    Register in Galaxy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-a-history" title="Import a history" class="md-nav__link">
    Import a history
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-reads-with-velvet" title="Assemble reads with Velvet" class="md-nav__link">
    Assemble reads with Velvet
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-the-output" title="Examine the output" class="md-nav__link">
    Examine the output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#collect-some-statistics-on-the-contigs" title="Collect some statistics on the contigs." class="md-nav__link">
    Collect some statistics on the contigs.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly-with-velvet-optimiser" title="Assembly with Velvet Optimiser" class="md-nav__link">
    Assembly with Velvet Optimiser
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#look-at-the-fasta-statistics-for-the-velvet-optimiser-contigs" title="Look at the fasta statistics for the Velvet Optimiser contigs" class="md-nav__link">
    Look at the fasta statistics for the Velvet Optimiser contigs
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly with Spades" class="md-nav__link">
      Genome assembly with Spades
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../variants/" title="Basic Variant finding" class="md-nav__link">
      Basic Variant finding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant finding" class="md-nav__link">
      Variant finding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Advanced Microbial Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Advanced Microbial Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../pacbio/" title="Assembly with SMRT portal" class="md-nav__link">
      Assembly with SMRT portal
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Microbial Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Microbial Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../dge/" title="Differential gene expression" class="md-nav__link">
      Differential gene expression
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../about/" title="About" class="md-nav__link">
      About
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#galaxy-background" title="Galaxy Background" class="md-nav__link">
    Galaxy Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#register-in-galaxy" title="Register in Galaxy" class="md-nav__link">
    Register in Galaxy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-a-history" title="Import a history" class="md-nav__link">
    Import a history
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assemble-reads-with-velvet" title="Assemble reads with Velvet" class="md-nav__link">
    Assemble reads with Velvet
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examine-the-output" title="Examine the output" class="md-nav__link">
    Examine the output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#collect-some-statistics-on-the-contigs" title="Collect some statistics on the contigs." class="md-nav__link">
    Collect some statistics on the contigs.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly-with-velvet-optimiser" title="Assembly with Velvet Optimiser" class="md-nav__link">
    Assembly with Velvet Optimiser
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#look-at-the-fasta-statistics-for-the-velvet-optimiser-contigs" title="Look at the fasta statistics for the Velvet Optimiser contigs" class="md-nav__link">
    Look at the fasta statistics for the Velvet Optimiser contigs
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="assembly-using-velvet">Assembly using Velvet</h1>
<h2 id="background">Background</h2>
<p>Velvet is one of a number of <em>de novo</em> assemblers that use short read sets as input (e.g. Illumina Reads), and the assembly method is based on de Bruijn graphs. For information about Velvet see this <a href="https://en.wikipedia.org/wiki/Velvet_assembler">link</a>.</p>
<!---
A protocol for assembling with Velvet (another *de novo* assembler) is available [here](https://docs.google.com/document/d/1xs-TI5MejQARqo0pcocGlymsXldwJbJII890gnmjI0o/pub).
--->

<p>In this activity, we will perform a <em>de novo</em> assembly of a short read set using the Velvet assembler.</p>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>assemble the reads using VelvetOptimiser, and</li>
<li>examine the output assembly.</li>
</ol>
<h2 id="galaxy-background">Galaxy Background</h2>
<p>Galaxy is a web-based analysis and workflow platform designed for biologists to analyse their own data. It can be used to run a variety of bioinformatics tools. The selection of bioinformatics tools installed on the Galaxy instance we are using today caters for the analysis of bacterial genomics data sets.</p>
<p>Galaxy is an open, web-based platform. Details about the project can be found <a href="https://galaxyproject.org/">here</a>.</p>
<p>The Galaxy interface is separated into three parts. The <ss>Tools</ss> list on the left, the <ss>Viewing</ss> panel in the middle and the analysis and data <ss>History</ss> on the right.</p>
<p><img alt="galaxy overview screenshot" src="images/galaxy-overview.png" /></p>
<h2 id="register-in-galaxy">Register in Galaxy</h2>
<p>Open a new tab or window on your web browser. Use Firefox or Chrome - please don’t use Internet Explorer or Safari.</p>
<p>In the address bar, type in the address of your galaxy server, e.g. <a href="https://usegalaxy.org.au">Galaxy Australia</a>.</p>
<p><img alt="Galaxy URL" src="images/galaxy_address_bar.png" /></p>
<p>Click on <ss>User</ss> button on the right.</p>
<p><img alt="Register or Login screenshot" src="images/location_of_user_menu.png" /></p>
<p>If you have never registered on this Galaxy server before:</p>
<ul>
<li>Select: <ss>User &rarr; Register</ss></li>
<li>Enter your email, choose a password, and choose a user name.</li>
<li>Click <ss>Submit</ss></li>
</ul>
<p>If you have, just login:</p>
<ul>
<li>Select: <ss>User &rarr; Login</ss></li>
<li>Enter your email and password.</li>
<li>Click <ss>Submit</ss></li>
</ul>
<p>Return to the home screen.</p>
<h2 id="import-a-history">Import a history</h2>
<ul>
<li>In the menu options across the top, go to <ss>Shared Data</ss>.</li>
<li>Click on <ss>Histories</ss>.</li>
</ul>
<p><img alt="Shared histories" src="images/shared_data_histories.png" /></p>
<ul>
<li>A list of published histories should appear. Click on the history called <strong>Microbial Genomics Workshop - BINF90002</strong></li>
<li>Click on <ss>Import history</ss>.</li>
<li>
<p>An option will appear to re-name the history. We don&rsquo;t need to rename it, so click <ss>Import</ss>.</p>
</li>
<li>
<p>The history will now appear in your Current History pane, and the files are ready to use in Galaxy analyses.</p>
</li>
<li>
<p>The read set for today is from an imaginary <em>Staphylococcus aureus</em> bacterium with a miniature genome.</p>
</li>
<li>
<p>The whole genome shotgun method used to sequence our mutant strain read set was produced on an Illumina DNA sequencing instrument.</p>
</li>
<li>
<p>The files we need for assembly are the <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>.</p>
</li>
<li>The reads are paired-end.</li>
<li>
<p>Each read is 150 bases long. <!--(before trimming)--></p>
</li>
<li>
<p>The number of bases sequenced is equivalent to 19x the genome sequence of the wildtype strain. (Read coverage 19x - rather low!).</p>
</li>
</ul>
<!--
- <fn>wildtype.fna</fn>: the reference genome sequence of the wildtype strain in fasta format (a header line, then the nucleotide sequence of the genome)

- <fn>wildtype.gff</fn>: the reference genome sequence of the wildtype strain in general feature format (a list of features - one feature per line, then the nucleotide sequence of the genome).

- <fn>wildtype.gbk</fn>: the reference genome sequence in genbank format.
--->

<ul>
<li>Click on the View Data button (the <img alt="Eye icon" src="images/image04.png" />) next to one of the FASTQ sequence files.</li>
</ul>
<!--
- The gff file should look like this:
- Brief Discussion about the GFF format (FIXME: add)
![GFF format](images/image08.png)

## Evaluate the input reads

Questions you might ask about your input reads include:

- How good is my read set?
- Do I need to ask for a new sequencing run?  
- Is it suitable for the analysis I need to do?

We will evaluate the input reads using the FastQC tool.

- This runs a standard series of tests on your read set and returns a relatively easy-to-interpret report.
- We will use the FastQC tool in Galaxy to evaluate the quality of one of our FASTQ files.
- Go to <ss>Tools &rarr; NGS:Analysis &rarr; NGS: QC and Manipulation &rarr; FastQC</ss>
- Select <fn>mutant_R1.fastq</fn>
- <ss>Execute</ss>
- Once finished, examine the output called <fn>FastQC on data1:webpage</fn> (Hint:![Eye icon](./images/image04.png)). It has a summary at the top of
the page and a number of graphs.

Some of the important outputs of FastQC for our purposes are:

-   <ss>Basic Statistics: Sequence length</ss>: will be important in setting maximum k-mer size value for assembly
-   <ss>Basic Statistics: Encoding</ss>: Quality encoding type: important for quality trimming software
-   <ss>Basic Statistics: % GC</ss>: high GC organisms don’t tend to assemble well and may have an uneven read coverage distribution.
-   <ss>Basic Statistics: Total sequences</ss>: Total number of reads: gives you an idea of coverage.
-   <ss>Per base sequence quality</ss>: Dips in quality near the beginning, middle or end of the reads: determines possible trimming/cleanup methods and parameters and may indicate technical problems with the sequencing process/machine run.
-   <ss>Per base N content</ss>: Presence of large numbers of Ns in reads: may point to poor quality sequencing run. You would need to trim these reads to remove Ns.
-   <ss>Kmer content</ss>: Presence of highly recurring k-mers: may point to contamination of reads with barcodes or adapter sequences.

Although we have warnings for two outputs (per base sequence content; Kmer content), we can ignore these for now. For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of ["good"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and ["bad"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data. We won’t be doing anything to these data to clean it up as there isn’t much need. Therefore we will get on with the assembly!

-->

<h2 id="assemble-reads-with-velvet">Assemble reads with Velvet</h2>
<p>Everyone will be assigned a value of k (k-mer length) to use in their assembly with Velvet. We will then populate a spreadsheet with result metrics from all of the different assemblies. The spreadsheet can be found <a href="https://docs.google.com/spreadsheets/d/1Ruik9-An8aPSVGIr5v29xyoGWpmq-I5iuRHfa-MPV6c/edit?usp=sharing">here</a>. Please put your name in a blank space in the <strong>Name</strong> column of the spreadsheet and note the value for k next to it.</p>
<ul>
<li>We will perform a <em>de novo</em> assembly of the mutant FASTQ reads into long contiguous sequences (in FASTA format.)</li>
<li>Velvet requires the user to input a value of <em>k</em> for the assembly process. K-mers are fragments of sequence reads. Small k-mers will give greater connectivity, but large k-mers will give better specificity.</li>
</ul>
<!---
- velvet produces both contigs and scaffolds.
Ask your demonstrator if you would like to know the difference between contigs and scaffolds.
--->

<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; velvet</ss></li>
<li>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li><ss>K-mer</ss>: Enter the value for k that you have been assigned in the spreadsheet.</li>
<li><ss>Input file type</ss>: Fastq</li>
<li><ss>Single or paired end reads</ss>: Paired</li>
<li><ss> Select first set of reads</ss>: <fn>mutant_R1.fastq</fn>  </li>
<li><ss> Select second set of reads</ss>: <fn>mutant_R2.fastq</fn></li>
</ul>
</li>
<li>
<p>Your tool interface should look something like this (you will most likely have a different value for k):</p>
</li>
</ul>
<p><img alt="velvet interface" src="images/image09.png" /></p>
<ul>
<li>Click <ss>Execute</ss></li>
</ul>
<h2 id="examine-the-output">Examine the output</h2>
<ul>
<li>Galaxy is now running velvet on the reads for you.</li>
<li>Press the refresh button in the history pane to see if it has finished.</li>
<li>
<p>When it is finished, you will have four new files in your history.  </p>
<ul>
<li>a <fn>Contigs</fn> file</li>
<li>a <fn>Contigs stats</fn> file</li>
<li>a <fn>LastGraph</fn> file</li>
<li>the velvet <fn>log</fn> file</li>
</ul>
</li>
<li>
<p>Click on the View Data button <img alt="Eye icon" src="images/image04.png" /> on each of the files.</p>
</li>
<li>
<p>The <fn>Contigs</fn> file will show each contig with the <em>k-mer length</em> and <em>k-mer coverage</em> listed as part of the header (however, these are just called <em>length</em> and <em>coverage</em>).</p>
<ul>
<li><em>K-mer length</em>: For the value of k chosen in the assembly, a measure of how many k-mers overlap (by 1 bp each overlap) to give this length.</li>
<li><em>K-mer coverage</em>: For the value of k chosen in the assembly, a measure of how many k-mers overlap each base position (in the assembly).</li>
</ul>
</li>
</ul>
<p><img alt="Contigs output" src="images/image10.png" /></p>
<ul>
<li>The <fn>Contigs stats</fn> file will show a list of these k-mer lengths and k-mer coverages.</li>
</ul>
<p><img alt="Contigs stats output" src="images/image11.png" /></p>
<ul>
<li>We will summarise the information in the <fn>log</fn> file.</li>
</ul>
<h2 id="collect-some-statistics-on-the-contigs">Collect some statistics on the contigs.</h2>
<ul>
<li>Go to <ss>NGS Common Toolsets &rarr; FASTA manipulation &rarr; Fasta statistics</ss></li>
<li>For the required input file, choose the velvet <fn>Contigs</fn> file.</li>
<li>Click <ss>Execute</ss>.</li>
<li>A new file will appear called <fn>Fasta summary stats</fn></li>
<li>Click the eye icon to look at this file. (It will look something like - but not exactly like - this.)</li>
</ul>
<p><img alt="Fasta stats" src="images/image12.png" /></p>
<ul>
<li>Look at:<ul>
<li><em>num_seq</em>: the number of contigs in the FASTA file.</li>
<li><em>num_bp</em>: the number of assembled bases. Roughly proportional to genome size.</li>
<li><em>len_max</em>: the biggest contig.  </li>
<li><em>len_N50</em>: N50 is a contig size. If contigs were ordered from small to large, half of all the nucleotides will be in contigs this size or larger.</li>
</ul>
</li>
</ul>
<p><strong>Now copy the relevant data back into the k-mer spreadsheet on your line.</strong></p>
<p>Along with the demonstrator, have a look at the effect of the k-mer size on the output metrics of the assembly. Note that there are local maxima and minima in the charts. What do you think is happening here? Why is the value of k (the k-mer size) having an effect? <!--In next week's lecture, this will be discussed in detail. --></p>
<h2 id="assembly-with-velvet-optimiser">Assembly with Velvet Optimiser</h2>
<p>Now that we have seen the effect of k-mer size on the assembly, we will run the Velvet Optimiser to automatically choose the best k-mer size for us. It will use the &ldquo;n50&rdquo; to determine the best k-mer value to use. It then performs the further graph cleaning steps and automatically chooses a bunch of other parameters for velvet. We should get a much better assembly result than we did with our attempts with Velvet alone..</p>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; Velvet Optimiser</ss></li>
<li>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li><ss>Start k-mer size</ss>: 45</li>
<li><ss>End k-mer size</ss>: 73</li>
<li><ss>Input file type</ss>: Fastq</li>
<li><ss>Single or paired end reads</ss>: Paired</li>
<li><ss> Select first set of reads</ss>: <fn>mutant_R1.fastq</fn>  </li>
<li>
<p><ss> Select second set of reads</ss>: <fn>mutant_R2.fastq</fn></p>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
</ul>
</li>
</ul>
<h2 id="look-at-the-fasta-statistics-for-the-velvet-optimiser-contigs">Look at the fasta statistics for the Velvet Optimiser contigs</h2>
<p>Use the Fasta Statistics tool you used earlier to summarise the Velvet Optimiser output. Examine the resulting table. What are the main differences?</p>
<p>Have a look at the Velvet Optimiser log file, it&rsquo;s hidden. Click on the <strong>hidden</strong> link at the top of the History pane. You&rsquo;ll then need to examine its STDERR output by clicking on the <strong>name of the file</strong>, then the <strong>&ldquo;i&rdquo;</strong> icon, then <strong>stderr</strong>.</p>
<p>Can you find which <strong>k</strong> value VelvetOptimiser used for its final assembly? You should also notice that it set another couple of parameters, the expected coverage and the coverage cutoff. Any ideas what these are? See the <a href="http://www.genome.org/cgi/content/short/18/5/821">Velvet paper</a> or the <a href="https://www.ebi.ac.uk/~zerbino/velvet/Manual.pdf">Velvet manual</a> for details on these parameters.</p>
<!-- ## What next?

- [Annotate the genome using Prokka.](/modules/prokka/index.md)
-->
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../data-dna/" title="Training dataset" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Training dataset
              </span>
            </div>
          </a>
        
        
          <a href="../spades/" title="Genome assembly with Spades" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Genome assembly with Spades
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.2ba8dec4.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
  </body>
</html>